include/sys_clock.h: simplify _NEED_PRECISE_TICK_MS_CONVERSION

gnuless opened the pull request:
This is just hygiene. Some preprocessor logic is optimized, eliminating
a temporary (_NON_OPTIMIZED_TICKS_PER_SEC) in the process.

Signed-off-by: Charles E. Youse <charles.youse@intel.com>

Commit:
include/sys_clock.h: simplify _NEED_PRECISE_TICK_MS_CONVERSION

This is just hygiene. Some preprocessor logic is optimized, eliminating
a temporary (_NON_OPTIMIZED_TICKS_PER_SEC) in the process.

Signed-off-by: Charles E. Youse <charles.youse@intel.com>

andrewboie commented:
recheck

andyross reviewed (approved):
Looks great.  While you're in there and if you wan to do more cleanup, "CONFIG_TIMER_READS_ITS_FREQUENCY_AT_RUNTIME" is there purely for the benefit of the HPET driver.  It might be replaced with a driver-specific override on the conversion or whatnot.

And the fixed modulus might be relaxed a little.  For fixed divisors, there are tricks to get "enough" precision by using the high bits of the computation instead of a simple division.

gnuless commented:
> And the fixed modulus might be relaxed a little. For fixed divisors, there are tricks to get "enough" precision by using the high bits of the computation instead of a simple division.

Are you talking about pre-scaling before the division or something?  Otherwise I'm not sure what you're getting at.  Sounds like a good lunch conversation.

Commit:
include/sys_clock.h: consolidate identical branches in __ticks_to_ms

The two branches of the compile-time conditional are identical, so
they are consolidated and the conditional removed.

Just hygiene again.  No functional change.

Signed-off-by: Charles E. Youse <charles.youse@intel.com>
